packages/@aws-cdk/aws-redshift/lib/cluster.ts

```diff
@@ -0,0 +1,413 @@
+import { ClusterAttributes, ICluster } from './cluster-ref';
+import { DatabaseSecret } from './database-secret';
+import { Endpoint } from './endpoint';
+import { IParameterGroup } from './parameter-group';
+import { Login, RotationMultiUserOptions } from './props';
+import { CfnCluster, CfnClusterSubnetGroup } from './redshift.generated';
+
+/**
+ * What node type to use
+ */
+export enum NodeType {
+    /**
+     *
+     */
+    DS2_XLARGE = "ds2.xlarge",
+    DS2_8XLARGE = "ds2.8xlarge",
+    DC1_LARGE = "dc1.large",
+    DC1_8XLARGE = "dc1.8xlarge",
+    DC2_LARGE = "dc2.large",
+    DC2_8XLARGE = "dc2.8xlarge",
+    RA3_16XLARGE = "ra3.16xlarge",
+}
+
+/**
+ * What cluster type to use
+ */
+export enum ClusterType {
+    SINGLE_NODE = "single-node",
+    MULTI_NODE = "multi-node"
+}
+
+/**
+ * Properties for a new database cluster
+ */
+export interface ClusterProps {
+
+    /**
+     * An optional identifier for the cluster
+     *
+     * @default - A name is automatically generated.
+     */
+    readonly clusterIdentifier?: string;
+
+    /**
+     * Additional parameters to pass to the database engine
+     * https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-parameter-groups.html
+     *
+     * @default - No parameter group.
+     */
+    readonly parameterGroup?: IParameterGroup;
+
+    /**
+     * Number of compute nodes in the cluster
+     *
+     * Value must be at least 1 and no more than 100.
+     *
+     * @default 1
+     */
+    readonly numberOfNodes?: number;
+
+    /**
+     * The node type to be provisioned for the cluster.
+     *
+     * @default NodeType.DC2_LARGE
+     */
+    readonly nodeType?: NodeType;
+
+    /**
+     * Settings for the individual instances that are launched
+     *
+     * @default ClusterType.MULTI_NODE
+     */
+    readonly clusterType?: ClusterType;
+
+    /**
+     * What port to listen on
+     *
+     * @default - The default for the engine is used.
+     */
+    readonly port?: number;
+
+    /**
+     * Whether to enable encryption of data in the cluster
+     *
+     * @default false
+     */
+    readonly encrypted?: boolean
+
+    /**
+     * The KMS key for storage encryption.
+     * will be set to `true`.
+     *
+     * @default - default master key.
+     */
+    readonly kmsKey?: kms.IKey;
+
+    /**
+     * A preferred maintenance window day/time range. Should be specified as a range ddd:hh24:mi-ddd:hh24:mi (24H Clock UTC).
+     *
+     * Example: 'Sun:23:45-Mon:00:15'
+     *
+     * @default - 30-minute window selected at random from an 8-hour block of time for
+     * each AWS Region, occurring on a random day of the week.
+     * @see https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/USER_UpgradeDBInstance.Maintenance.html#Concepts.DBMaintenance
+     */
+    readonly preferredMaintenanceWindow?: string;
+
+    /**
+     * The VPC to place the cluster in.
+     *
+     * @default a new Vpc will be created
```

Comment:
1. I appreciate the sentiment, I really do. But the VPC is so crucial, that we always make it a required property of constructs that use it.
2. Whenever VPC is a construction property, you also need a `vpcSubnets?: ec2.SubnetSelection` property to go with it.

Reply:
Cool. Didn't know that, but makes a lot of sense.